Разметка внутренней страницы

liketurbo opened the pull request:
---
:mortar_board: [Разметка главной страницы](https://up.htmlacademy.ru/htmlcss/33/user/2006355/tasks/3)

---
:mortar_board: [Разметка внутренней страницы](https://up.htmlacademy.ru/htmlcss/33/user/2006355/tasks/4)

:boom: https://htmlacademy-htmlcss.github.io/2006355-device-33/2/

liketurbo committed:
feat: add markup for index page

liketurbo committed:
chore: update comment

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        class="search-form"
+        action="https://echo.htmlacademy.ru/"
+        method="get"
+      >
```
В верстке не принято переносить атрибуты на новую строчку: https://codeguide.academy/html-css.html#html-tags-formatting. Код раздувается, тяжелее читать.

liketurbo replied:
Снес prettier, проблема ушла

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+    <!-- Главный хедер страницы -->
+    <header class="page-header">
+      <h1 class="visually-hidden">Device</h1>
+      <img class="logo" src="" alt='Логотип "Device"' />
```
Кавычки в атрибутах везде двойные можно использовать. https://codeguide.academy/html-css.html#html-attr-double-quotes
<img class="logo" src="" alt="Логотип Барбершопа «Бородинский»."> В лекции был такой пример, можно кавычки сделать типа «таких»

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+            <a class="service-nav-link" href="#">Корзина</a>
+          </li>
+        </ul>
+      </nav>
```
Я бы здесь разделила элементы: Войти отдельно, Сравнить -- отдельно и Корзина -- отдельно. Список -- это то, что связано между собой, например, список элементов меню, список карточек товаров. А здесь нет логической составляющей списка.

liketurbo replied:
Вынес как отдельные ссылки

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        Каталог товаров
+      </button>
+      <nav class="dropdown-nav visually-hidden" id="nav-dropdown-list">
+        <ul class="dropdown-list">
```
Я думаю, здесь достаточно сделать иконку плюсика и все. Выпадающий список не нужно будет делать

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        aria-controls="nav-dropdown-list"
+      >
+        Каталог товаров
+      </button>
```
Это не кнопка, а ссылка на страницу Каталога. Иконка плюса может быть кнопкой с выпадающим списком, но Каталог я бы сделала ссылкой.

liketurbo replied:
Разделил кнопку и ссылку

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        <!-- Описание особенности -->
+        <div class="promo-desc">
+          <span>01</span>
+          <p class="promo-title">Порхает как бабочка, жалит как пчела!</p>
```
Выглядит как заголовок, я бы и сделала это заголовком

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+          <p class="promo-content">
+            Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером,
+            замаскированным под стандартную камеру.
+          </p>
```
https://codeguide.academy/html-css.html#html-tags-formatting -- Текст внутри тегов остаётся на одной строке с тегами

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        <a class="special-delivery-button" href="#">
+          <span class="visually-hidden">Узнать больше</span>
+        </a>
+      </article>
```
сюда section подойдет лучше. article -- отделяемый от сайта смысловой раздел. А здесь даже не смысловой раздел, а секция или блок с контентом

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+          <h2 class="accordion-title">Кредит</h2>
+        </section>
+      </div>
+
```
Здесь немножко по семантике странно выходит: ты вкладываешь секции в див, но по логике, все это большая секция -- accordion-wrapper, в которую входят дивы accordion-section, их на самом деле можно назвать accordion-item

liketurbo replied:
Заменил section(ы) на лист

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+          action="https://echo.htmlacademy.ru/"
+          method="post"
+        >
+          <input class="newsletter-input" name="email" type="email" required />
```
В input еще можно добавить атрибут placeholder="Ваш e-mail", это подсказка для пользователя

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+      <!-- Адрес офиса -->
+      <section class="footer-address">
+        <h2 class="visually-hidden">Адрес</h2>
+        <img class="logo logo-alt" src="" alt='Логотип "Device"' />
```
Здесь такая же коммент, как выше, к хедеру

liketurbo replied:
Нашел хорошую статью по кавычкам - [§ 104. Кавычки - Студия Артемия Лебедева](https://www.artlebedev.ru/kovodstvo/sections/104/)

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
+        <h2 class="visually-hidden">Контакты</h2>
+        <a class="footer-phone" href="tel:+78128121212">+7 812 812-12-12</a>
+        <a class="academy-link" href="https://htmlacademy.ru/">
+          <img class="academy-logo" src="" alt='Логотип "HTML Academy"' />
```
<img class="logo" src="" alt="Логотип Барбершопа «Бородинский»."> В лекции был такой пример, можно кавычки сделать типа «таких»

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,340 @@
 <html lang="ru">
   <head>
-    <meta charset="utf-8">
+    <meta charset="utf-8" />
```
Без слеша: https://codeguide.academy/html-css.html#html-closed-tag

liketurbo replied:
Ох, уж этот prettier. Опять)

ndeg-web reviewed (changes requested):
Рамзан, в целом хорошая разметка. Есть несколько однотипных недочетов и неточностей, я о них написала. Поправь их, пожалуйста, и можно идти  дальше. 
Понравилось, что ты написал role и написал комментарии к коду, это хорошая практика.

liketurbo committed:
refac: change default html formatter

Subsequently, it fixes:
- https://codeguide.academy/html-css.html#html-tags-formatting
- https://codeguide.academy/html-css.html#html-closed-tag

liketurbo committed:
fix: add forgotten class

liketurbo committed:
feat: add input placeholders

liketurbo committed:
fix: set proper quotation marks

More info: https://www.artlebedev.ru/kovodstvo/sections/104/

liketurbo committed:
fix: change button to link

liketurbo committed:
refac: change "service links" layout

liketurbo committed:
feat: add catalog page

liketurbo committed:
refac: change tags for "features accordion"

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+        <li class="accordion-select-item" aria-controls="accordion-section-2">
+          <button class="accordion-select-button" type="button">Гарантия</button>
+        </li>
+        <li class="accordion-select-item" aria-controls="accordion-section-2">
```
Тут опечатка )) `accordion-section-3`

liketurbo replied:
Пипец, вы хорошо заметитили эту опечатку)

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+          <h3 class="accordion-preview-title visually-hidden">Гарантия</h3>
+        </li>
+        <li class="accordion-preview-item visually-hidden" role="tabpanel" id="accordion-preview-item-3">
+          <h3 class="accordion-title">Кредит</h3>
```
Здесь тоже наверное нужен класс ` visually-hidden`, как в Гарантии

liketurbo replied:
Если родитель уже visually-hidden, то детям тоже надо ставить этот класс?

ndeg-web replied:
Нет, дети все будут скрыты, так как скрыт родитель

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+    </nav>
+
+    <!-- Аккордеон особенностей -->
+    <div class="features-accordion">
```
мне кажется, здесь и `section `хорошо пойдет

liketurbo replied:
Ахх, я переделывал как раз, чтобы сделать section и забыл изменить tag

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+    </div>
+
+    <!-- Доставка редкого товара -->
+    <article class="special-delivery">
```
Тут лучше будет `section`

liketurbo replied:
Тут я сомневался насчет section(а), потому что из текста тут только заголовок

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+    </div>
+
+    <!-- Ссылки на страницу сравнения и корзину -->
+    <a class="small-link compare-link" href="#">Сравнить</a>
```
Не заметила в прошлый раз, тут нужна еще иконка сравнения

liketurbo replied:
Добавил svg из макета

ndeg-web commented on index.html:
```diff
@@ -1,12 +1,301 @@
+
+    <!-- Ссылки на страницу сравнения и корзину -->
+    <a class="small-link compare-link" href="#">Сравнить</a>
+    <a class="small-link cart-link" href="#">Корзина</a>
```
А тут иконка корзины и количество товаров в корзине

liketurbo replied:
Добавил

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,284 @@
+            <ul class="catalog-filter-list">
+              <li class="catalog-filter-item">
+                <label class="control">
+                  Черный <input class="control-input" type="checkbox" name="black" checked>
```
Тут наверное надо местами поменять, в макете сначала чекбокс, а затем надпись к нему

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,284 @@
+
+      <!-- Обертка для продуктов -->
+      <div class="products-wrapper">
+        <h3 class="visually-hidden">Сами продукты</h3>
```
Может быть Список продуктов лучше будет?

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,284 @@
+      <address class="footer-address">
+        Санкт-Петербург, набережная реки Карповки, 5, литера П.
+      </address>
+      <hr class="dash dash-alt footer-address-dash">
```
Ой, как много классов на один hr )) Можно оставить какой-нибудь один

liketurbo replied:
Оставил max два класса

ndeg-web reviewed (changes requested):
Рамзан, хорошо получилось!
Есть несколько замечаний, поправь их, пожалуйста, и можно идти дальше.

liketurbo committed:
fix: change div to section

liketurbo committed:
fix: changes by the last review

liketurbo committed:
feat: add icons to catalog header

liketurbo committed:
fix: update according to last review

liketurbo committed:
fix: remove excessive amout of classes

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,329 @@
+        <path
+          d="M15.7 14.6889L12.6 11.5889C13.5 10.3889 14 8.98892 14 7.38892C14 3.48892 10.9 0.388916 7 0.388916C3.1 0.388916 0 3.48892 0 7.38892C0 11.2889 3.1 14.3889 7 14.3889C8.6 14.3889 10 13.8889 11.2 12.9889L14.3 16.0889C14.5 16.2889 14.7 16.3889 15 16.3889C15.6 16.3889 16 15.9889 16 15.3889C16 15.1889 15.9 14.8889 15.7 14.6889ZM2 7.38892C2 4.58892 4.2 2.38892 7 2.38892C9.8 2.38892 12 4.58892 ...
+          fill="black" />
+      </svg>
```
Рамзан, не спеши изображения вставлять в код, я бы отнесла это к декоративным изображениям и убрала бы в стили, так и код разметки чище, и правильно методологически.

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,329 @@
+          d="M15.7 14.6889L12.6 11.5889C13.5 10.3889 14 8.98892 14 7.38892C14 3.48892 10.9 0.388916 7 0.388916C3.1 0.388916 0 3.48892 0 7.38892C0 11.2889 3.1 14.3889 7 14.3889C8.6 14.3889 10 13.8889 11.2 12.9889L14.3 16.0889C14.5 16.2889 14.7 16.3889 15 16.3889C15.6 16.3889 16 15.9889 16 15.3889C16 15.1889 15.9 14.8889 15.7 14.6889ZM2 7.38892C2 4.58892 4.2 2.38892 7 2.38892C9.8 2.38892 12 4.58892 ...
+          fill="black" />
+      </svg>
+      <input class="search-input" name="search" type="search" placeholder="Текст поиска" required>
```
Нужно `<input>` обернуть в `<label>`, см. [Критерий](https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-05)

liketurbo replied:
Стоит ли оборачивать `<input>` в `<label>` при отсутствии подписи?

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,329 @@
+            d="M0.5 12.3889H13.5C13.3 9.78892 11.4 7.58892 8.8 6.88892C9.8 6.28892 10.5 5.18892 10.5 3.88892C10.5 1.98892 8.9 0.388916 7 0.388916C5.1 0.388916 3.5 1.98892 3.5 3.88892C3.5 5.18892 4.2 6.28892 5.2 6.88892C2.6 7.58892 0.7 9.78892 0.5 12.3889Z"
+            fill="black" />
+        </svg>
+
```
Тут и ниже -- думаю, что стоит убрать это в стили (Если не слушал лекцию по Графике и базовой стилизации), то пока удали эти свг. В третьей лекции разбирают, что делать с ними.

ndeg-web commented on catalog.html:
```diff
@@ -0,0 +1,329 @@
+        </svg>
+        <span class="visually-hidden">Скрыть / Раскрыть</span>
+      </button>
+    </div>
```
Смотри, в макете есть развернутое состояние пунктов меню Каталога 
[Ссылка](https://www.figma.com/file/f8ZtnZcdWC2pSklG1ssm9K/HTML-1-%2F-%D0%94%D0%B5%D0%B2%D0%B0%D0%B9%D1%81?node-id=11534%3A6989)

ndeg-web reviewed (approved):
Рамзан, все хорошо!
Написала пару комментов, они не критичные, поправь, пожалуйста, в следующих коммитах. 
Можно двигаться дальше.
